src/CanvasChart.js

```diff
@@ -348,24 +369,37 @@ export default function CanvasChart({
       const maxX =
         firstTask.month * monthWidth - MARGIN - LINE_SEGMENT_MIN_LENGTH;
       const minX = parentTask.month * monthWidth + MARGIN;
-      const x = Math.max(
-        minX,
-        Math.min(
-          maxX,
-          parentTask.month * monthWidth + (parentTask.length * monthWidth) / 2
-        )
-      );
+      const x = !parentTask.isMilestone
+        ? Math.max(
+            minX,
+            Math.min(
+              maxX,
+              parentTask.month * monthWidth +
+                (parentTask.length * monthWidth) / 2
+            )
+          )
+        : parentTask.month * monthWidth + MILESTONE_SIZE / 2 - LINE_WIDTH / 2;
+
+      // If this is a milestone, adjust the vertical line verticies accordingly.
+      const milestoneVLineAdjustment0 = parentTask.isMilestone
+        ? 0 - MARGIN - MARGIN - MARGIN
```

Comment:
Can you explain why this line makes sense? Looks kind of suspicious to me.

Reply:
It actually is removing three margins, since those are baked into the row heights. I had to strip out extra margins to the center the vertex in the channel. Obviously could refactor the math itself.

Generally speaking, I agree, the entire layout section the way I have altered it could be refactored, but it was changing as I wrote the code and I didn't want to mess with your core work. There are already conflicts, and that's on me for not checking with you about if you wanted to bake in milestones.

Depending on if you want to proceed with the milestone functionality (and project altogether!) I think defining a "lane" concept would be useful (that is, basically, a row-height independent of the Avatar/Bar height.) This would make layout math much cleaner and easier to read.

Reply:
Update--the conflict seems to be easily resolved